unittests/runtime/Mutex.cpp

```diff
@@ -13,12 +13,57 @@
 #include "swift/Runtime/Mutex.h"
 #include "gtest/gtest.h"
 #include <chrono>
+#include <map>
+#include <random>
 #include <thread>
 
 using namespace swift;
 
+// When true many of the threaded tests log activity to help triage issues.
+static bool trace = false;
+
+// Note the use of non-atomic booleans for "lazy" thread control in many of
+// the tests is sufficient for what we are doing here, no need for
+// atomic operations.
+
+template <typename ThreadBody, typename AfterSpinRelease>
+void threadedExecute(int threadCount, ThreadBody threadBody,
+                     AfterSpinRelease afterSpinRelease) {
+
+  std::vector<std::thread> threads;
+
+  // Block the threads we are about to create.
+  bool spinWait = true;
```

Comment:
The way the tests work we can be lazy about the atomic aspects of things. It will get the correct answer soon enough across all affected threads. I can change if folks feel that strongly.

I put a note a little higher up in the tests attempting to outline why I was using atomic operations.